From 3cd5b95ac4ae68eb657f053336cbe9a9ab415fe3 Mon Sep 17 00:00:00 2001
From: Viorel Suman <viorel.suman@nxp.com>
Date: Thu, 14 Dec 2017 14:03:51 +0200
Subject: [PATCH 3095/5242] MLK-17220: ASoC: fsl_rpmsg_i2s: restore original
 lock context

commit  83026d03b7187652e7303f785d882b73e7266252 from
https://source.codeaurora.org/external/imx/linux-imx.git

Restore original lock context and unlock the mutex in case if
info->rpdev is uninitialized.

Signed-off-by: Viorel Suman <viorel.suman@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_rpmsg_i2s.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/fsl/fsl_rpmsg_i2s.c b/sound/soc/fsl/fsl_rpmsg_i2s.c
index 95cbe45..60b872f 100644
--- a/sound/soc/fsl/fsl_rpmsg_i2s.c
+++ b/sound/soc/fsl/fsl_rpmsg_i2s.c
@@ -35,15 +35,15 @@ static int i2s_send_message(struct i2s_rpmsg_s *msg,
 {
 	int err;
 
+	mutex_lock(&info->tx_lock);
 	if (!info->rpdev) {
 		dev_dbg(info->dev, "rpmsg channel not ready, m4 image ready?\n");
+		mutex_unlock(&info->tx_lock);
 		return -EINVAL;
 	}
 
 	dev_dbg(&info->rpdev->dev, "send cmd %d\n", msg->header.cmd);
 
-	mutex_lock(&info->tx_lock);
-
 	reinit_completion(&info->cmd_complete);
 	err = rpmsg_send(info->rpdev->ept, (void *)msg,
 			 sizeof(struct i2s_rpmsg_s));
-- 
1.7.9.5

